keanu-project/src/main/java/io/improbable/keanu/algorithms/mcmc/SamplingAlgorithm.java

```diff
@@ -12,6 +12,7 @@
 
     static Map<VariableReference, ?> takeSample(List<? extends Variable> sampleFromVariables) {
         return sampleFromVariables.stream()
```

Comment:
Given we call this per sample, are we happy we're adding cost for a case of "User Error"?  Could we filter this once at the beginning of of the sampling run?

Reply:
Maybe make it a Set? We can do this distinct on construction at the very least.